assets/js/aplus.js

```diff
@@ -665,28 +669,22 @@ $(function() {
  * to the next
  * Note! Translation requires that at least 'hl' query parameter is copied
  */
-(function(window) {
-    window.addEventListener('click', function(event) {
-        if (
-            event.target.nodeName === 'A'
-            && event.target.href
-            && event.target.protocol === window.location.protocol
-            && event.target.host === window.location.host // hostname:port
-            && (event.target.search === '' || event.target.search === '?') // no search
-        ) {
-            event.preventDefault(); // stop default redirect
-            event.target.search = window.location.search; // copy query params
-            if (event.target.getAttribute('target') === '_blank') {
-                // Open a new window or tab.
-                window.open(
-                    event.target.href,
-                    '_blank',
-                    'location,menubar,resizable,scrollbars,status,noopener');
-            } else {
-                window.location.href = event.target.href; // redirect the current window
-            }
-        }
-    }, false);
+(function (window) {
+  // The following regex takes the language values from the LANGUAGES variable in the aplus/settings.py, i.e., en and fi.
+  // Therefore, the values must be adjusted accordingly.
+  const regexp = /\?+.*hl=(en|fi)+/gi;
+  const addQueryParameter = (event) => {
+    if (
+      event.target.protocol === window.location.protocol &&
+      event.target.host === window.location.host && // hostname:port
+      window.location.search.match(regexp) // hl search
+    ) {
+      event.target.search = window.location.search; // copy query params
+    }
+  };
+
+  // Add event to all the <a> tags with a href attribute.
+  $(document).on("click", "a[href]", addQueryParameter);
```

Comment:
To my understanding `jquery.on(event, filter)` is identical to old implementation in performance point of view. Both should capture all click events and do check if the filter match. In old case, we checked if the target object is `a` tag and in new one there is a jquery filter. In a new implementation, our code is not executed for every click, but jquery is. It's possible that I'm wrong and web browsers have an API to do the filtering before JS code.

Reply:
My idea was to use the `jquery.on()` function to create a delegate event handler. 

>Delegated event handlers have the advantage that they can process events from descendant elements that are added to the document at a later time.

https://api.jquery.com/on/#direct-and-delegated-events